stopcovid/dialog/models/events.py

```diff
@@ -415,9 +419,104 @@ def __init__(self, phone_number: str, user_profile: UserProfile, **kwargs):
         )
 
     def apply_to(self, dialog_state: DialogState):
+        dialog_state.current_drill = None
+        dialog_state.drill_instance_id = None
+        dialog_state.current_prompt_state = None
         dialog_state.user_profile.opted_out = False
 
 
+class NameChangeDrillRequestedSchema(DialogEventSchema):
+    @post_load
+    def to_dataclass(self, data, **kwargs):
+        return NameChangeDrillRequested(**{k: v for k, v in data.items() if k != "event_type"})
+
+
+class NameChangeDrillRequested(DialogEvent):
+    def __init__(self, phone_number: str, user_profile: UserProfile, **kwargs):
+        super().__init__(
+            NameChangeDrillRequestedSchema(),
+            DialogEventType.NAME_CHANGE_DRILL_REQUESTED,
+            phone_number,
+            user_profile,
+            **kwargs,
+        )
+
+    def apply_to(self, dialog_state: DialogState):
+        dialog_state.current_drill = None
+        dialog_state.drill_instance_id = None
+        dialog_state.current_prompt_state = None
+        dialog_state.user_profile.opted_out = False
```

Comment:
Should this count for opting someone back in? Maybe we should remove it from schedule too

Reply:
yeah I think it makes sense that if you opt out, but then later continue to interact with the system, that you're opted back in. But I could be convinced otherwise!

Reply:
This is good for now. I need to read more about compliance